### PR TITLE
[ONNX] Export Eq, Ne: Cast input types

### PR DESCRIPTION
**Pull request by neginraoof**
This allows us to compare castable types. It'll fix:
https://github.com/pytorch/pytorch/issues/26189

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -775,12 +775,14 @@ def wrap_with_not(g, input, other):
 
 
 def eq(g, self, other):
-    return g.op("Equal", self, other)
+    other = sym_help._maybe_get_scalar(other)
```
Comment (lara-hdr) [inline torch/onnx/symbolic_opset9.py]: add tests for this particular case?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -775,12 +775,14 @@ def wrap_with_not(g, input, other):
...
+    return g.op("Equal", self, sym_help._if_scalar_type_as(g, other, self))
 
 
 @wrap_logical_op_with_negation
 def ne(g, self, other):
-    return g.op("Equal", self, other)
+    other = sym_help._maybe_get_scalar(other)
```
Comment (houseroad) [inline torch/onnx/symbolic_opset9.py]: just call eq?

**Review by houseroad — COMMENTED** [review]
With ScalarTypeAnalysis pass, is this change still necessary? cc: @BowenBao

**Comment by BowenBao** [comment]
> With ScalarTypeAnalysis pass, is this change still necessary? cc: @BowenBao

No, this change is not needed anymore.

**Comment by neginraoof** [comment]
Thanks @BowenBao!

**Pull request closed without merging**